File: Medium/Problem-1894/solution.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        total_chalk = sum(chalk)
        resto = k % total_chalk
        for i, x in enumerate(chalk):
            if(resto < x):
                return i
            resto -= x
        return 0

File: Medium/Problem-1894/Solution.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long total_chalk = 0;
        
        for (int i : chalk) {
            total_chalk += i;
        }

        long resto = k % total_chalk;

        for (int i = 0; i < chalk.length; i++) {
            if (resto < chalk[i]) {
                return i;
            }
            resto -= chalk[i];
        }
        
        return 0;

    }
}